ui/app/components/policy-editor.js

```diff
@@ -22,11 +22,18 @@ export default class PolicyEditorComponent extends Component {
     try {
       const nameRegex = '^[a-zA-Z0-9-]{1,128}$';
       if (!this.policy.name?.match(nameRegex)) {
-        throw new Error(
-          'Policy name must be 1-128 characters long and can only contain letters, numbers, and dashes.'
-        );
+        throw {
+          errors: [
+            {
+              detail:
+                'Policy name must be 1-128 characters long and can only contain letters, numbers, and dashes.',
+            },
+          ],
+        };
       }
 
+      const shouldRedirectAfterSave = this.policy.isNew;
```

Comment:
`question`: Looks we're shaping this error message to match the schema of an `AdapterError`. Are we just doing this because we don't want to update the utility function `messageError` that we have?

If `ember-data` updates the schema of the `AdapterError` then we'll have a pretty challenging situation to understand if we broke something.

Could we patch `messageError` to work with errors that don't stem from an HTTP response because I imagine we'll throw more errors in components because we've just recently started using flash messages?